units/check_pprof.go

```diff
@@ -0,0 +1,118 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"runtime/pprof"
+	"time"
+
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	pprofCheckName     = "pprof-check"
+	goroutineThreshold = 1500
+	pprofPath          = "/srv/evergreen/pprof"
+	totalBytesLimit    = 1000000000
+	checkInterval      = 5 * time.Minute
+)
+
+func init() {
+	registry.AddJobType(pprofCheckName,
+		func() amboy.Job { return makePprofCheckJob() })
+}
+
+type pprofCheckJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+}
+
+func makePprofCheckJob() *pprofCheckJob {
+	j := &pprofCheckJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    pprofCheckName,
+				Version: 0,
+			},
+		},
+	}
+	return j
+}
+
+func NewPprofCheckJob(ts string) amboy.Job {
+	j := makePprofCheckJob()
+	j.SetID(fmt.Sprintf("%s.%s", pprofCheckName, ts))
+
+	return j
+}
+
+func (j *pprofCheckJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	goroutineCount := runtime.NumGoroutine()
+	if goroutineCount < goroutineThreshold {
+		return
+	}
+
+	if err := os.MkdirAll(pprofPath, os.ModePerm); err != nil {
+		j.AddError(errors.Wrapf(err, "creating pprof path '%s'", pprofPath))
+		return
+	}
+
+	contents, err := ioutil.ReadDir(pprofPath)
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "listing contents of '%s'", pprofPath))
+		return
+	}
+
+	var totalSize int64
+	for _, file := range contents {
+		age := time.Now().Sub(file.ModTime())
+		if age < checkInterval {
+			grip.Info(message.Fields{
+				"message":            "not collecting pprof",
+				"reason":             "most recent file is newer then the threshold",
+				"mostRecentFile":     file.Name(),
```

Comment:
nit: it doesn't matter that much since this is a temporary piece of code, but snake case is more common for these JSON messages.

Reply:
done